### ISSUE TITLE
Assertion failure when loading emoji-only font

### ISSUE THREAD

**Issue by Ristovski** [ocornut/imgui#4566]
**Build info/versions**

<details>
<summary>Click to expand</summary>

```
Dear ImGui 1.85 WIP (18416)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=201402
define: __linux__
define: __GNUC__=10
--------------------------------
io.BackendPlatformName: imgui_impl_glfw
io.BackendRendererName: imgui_impl_opengl3
io.ConfigFlags: 0x00000000
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000000E
 HasMouseCursors
 HasSetMousePos
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 667.00,736.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

System info:
 - Gentoo Linux
 - freetype2 2.11
</details>

**My Issue/Question:**

Currently it does not appear to be possible to load emoji-only fonts such as the popular `Noto Color Emoji` (which can be downloaded from [`googlefonts`](https://github.com/googlefonts/noto-emoji/tree/main/fonts) repo [here](https://raw.githubusercontent.com/googlefonts/noto-emoji/main/fonts/NotoColorEmoji_WindowsCompatible.ttf)).

Loading such font generates the following error at runtime: 
```
example_glfw_opengl3: ../../imgui.h:1727: T& ImVector<T>::back() [with T = ImFontGlyph]: Assertion `Size > 0' failed.
Aborted
```

Loading the font into `ImGuiFontStudio` with the `freetype2` rasterizer shows "N Glyphs: 0", which basically conveys the same info as the error above (the `stb` one shows 200-ish glyphs but fails to render anything).

However the font itself works fine in all other applications, _**including the `ftview` utility from `freetype2`**_: 

![image](https://user-images.githubusercontent.com/994445/134093541-bfc61596-5acf-4cb4-8ee8-f9ffa41bce85.png)

this _should_ mean this is not a freetype2 issue, but rather imgui not handling this ("special") font properly.

EDIT: Just noticed that in `ftview` they show up differently (top right): `NotoColorEmoji` shows up as **`truetype bitmap`**, but `seguiemj` shows up as **`truetype v40`**.

The font seems to have a fixed size of `109` which can not be changed (not even in `ftview`), but setting that as the size for imgui to load does not resolve the issue.

I suspect that imgui does not handle fonts like this one, but I am not sure and haven't check the imgui_freetype code much yet, any ideas/help very appreciated.

> _I understand this is probably a very low-prio issue as virtually no one would load an emoji-only font into imgui, but this **might** be uncovering a potential future issue._

PS: Segoe UI Emoji works fine, but that one has regular glyphs as well. `WindowsCompatible` version of `NotoColorEmoji` is used deliberately for anyone wishing to test on Windows (both versions seem to work the same on Linux).

**Standalone, minimal, complete and verifiable example:** 
```c++
// Compile with -DIMGUI_USE_WCHAR32 -DIMGUI_ENABLE_FREETYPE
// Note the 'cfg.MergeMode = false', setting this to true would prevent the crash
// from occurring since it would try to merge 0 glyphs - essentially not doing anything.
io.Fonts->AddFontDefault();
{
    static ImWchar ranges[] = { 0x1, 0x1FFFF, 0 };
    static ImFontConfig cfg;
    cfg.OversampleH = cfg.OversampleV = 1;
    cfg.MergeMode = false;
    cfg.FontBuilderFlags |= ImGuiFreeTypeBuilderFlags_LoadColor;
    io.Fonts->AddFontFromFileTTF("/tmp/NotoColorEmoji_WindowsCompatible.ttf", 109.0f, &cfg, ranges);
    io.Fonts->Build();
}
```

**Comment by ocornut** [comment]
I don't think this has anything to do with the font being "emoji only" it's just that there's a quintillion of ways colored emojis can be encoded.

That specific one assert -we can fix- is probably happening when zero glyph have been loaded. The fact that zero glyphs have been loaded needs to be investigated. If you are interested in using this font could you investigate it?

**Comment by ocornut** [comment]
I'm not sure you are hitting the same bug, but note that according to #4567 it looks like freetype 2.11 has a bug with bitmap rendering.

**Comment by rokups** [comment]
![image](https://user-images.githubusercontent.com/19151258/134328689-3416326d-85a2-42ff-9779-ee135828c924.png)
I can confirm that this is freetype bug mentioned [here](https://github.com/ocornut/imgui/issues/4567#issuecomment-924344144). Applying patch fixes rendering. However note that `NotoColorEmoji_WindowsCompatible.ttf` is not a scalable font, so using any other size than size of embedded bitmaps will fail. Size of this particular font is 109.

**Comment by ocornut** [comment]
Thanks!
Closing this as a Freetype bug then.

Next commit will add a comment in imgui_freetype.cpp to help users navigate their way out of this issue:

```cpp
// If this crash for you: FreeType 2.11.0 has a crash bug on some bitmap/colored fonts.
// - https://gitlab.freedesktop.org/freetype/freetype/-/issues/1076
// - https://github.com/ocornut/imgui/issues/4567
// - https://github.com/ocornut/imgui/issues/4566
// You can use FreeType 2.10, or the patched version of 2.11.0 in VcPkg, or probably any upcoming FreeType version.
FT_Error error = FT_Load_Glyph(Face, glyph_index, LoadFlags);
if (error)
    return NULL;
```
